src/model/table.py

```diff
@@ -20,8 +28,57 @@ def set(self):
     def delete(self):
         pass
 
-    def get(self):
-        pass
+    def get(self, query):
+        efficient_index = self.__get_efficient_index__(query)
+        if efficient_index is None:
+            efficient_keys = self.__get_all_primary_keys__()
+        else:
+            efficient_keys = efficient_index.get_primary_keys(query[efficient_index.name])
+        found_objects = self.__get_rows__(efficient_keys)
+        return self.__filter_by_query__(found_objects, query)
 
-    def get_by_primary_key(self):
-        pass
+    def __get_efficient_index__(self, query):
+        if not query or str(query).isspace:
+            return None
+        if self.__table_metadata__.primary_key in query.keys():
+            return Index(self.__table_metadata__.primary_key, self.__table_metadata__)
+        index_names = set(self.__table_metadata__.get_indices_names())
+        most_efficient = None
+        for item in query.keys():
+            if item in index_names:
+                current_index = self.__table_metadata__.get_index(item)
+                if most_efficient is None or current_index.compare(most_efficient) == -1:
+                    most_efficient = current_index
+        return most_efficient
+
+    def __get_all_primary_keys__(self):
+        primary_keys = []
+        for primary_key in os.listdir(self.__path__):
+            if ".json" in str(primary_key) and "schema" not in str(primary_key):
+                primary_keys.append(self.get_by_primary_key(str(primary_key).replace(".json", '')))
+        return primary_keys
+
+    def get_by_primary_key(self, primary_key):
+        path = os.path.join(self.__path__, "{}.json".format(primary_key))
+        if not os.path.exists(path):
+            return None
+        with open(path, 'r') as file:
+            return json.load(file)
+
+    @staticmethod
+    def __filter_by_query__(found_objects, query):
+        if query and not str(query).isspace:
```

Comment:
not good code 

```
list = []
if query and not str(query).isspace:
    return list

for object_to_compare in found_objects:
     if (Tables.compare(object_to_compare, query))
         list.add(object_to_compare)
```

It isn't good to change in the data of the given input. This is very very bad